### ISSUE TITLE
Maven visitors should not be visiting XML documents without a Maven marker

### ISSUE THREAD

**Issue by mxhc** [openrewrite/rewrite-maven-plugin#444]
Hello, I saw that this issue was mentioned in https://github.com/openrewrite/rewrite-maven-plugin/issues/297
and that is closed. 
But I'm having it again with newer version of rewrite-maven-plugin - 4.37.0

```[INFO] <<< rewrite-maven-plugin:4.37.0:run (default-cli) < compile @ ebank_engine_payments-service <<<
[INFO] 
[INFO] 
[INFO] --- rewrite-maven-plugin:4.37.0:run (default-cli) @ ebank_engine_payments-service ---
[INFO] Using active recipe(s) [org.openrewrite.java.spring.boot2.UpgradeSpringBoot_2_7]
[INFO] Using active styles(s) []
[INFO] Validating active recipes...
[INFO] Project [ebank_engine_payments-service] Resolving Poms...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  16.894 s
[INFO] Finished at: 2022-12-01T11:00:42+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:4.37.0:run (default-cli) on project ebank_engine_payments-service: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:4.37.0:run failed: java.lang.IllegalStateException: Maven visitors should not be visiting XML documents without a Maven marker -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

Process finished with exit code 1
```

those are the versions that I used:

`mvn org.openrewrite.maven:rewrite-maven-plugin:4.37.0:run -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-spring:4.30.0  -DactiveRecipes=org.openrewrite.java.spring.boot2.UpgradeSpringBoot_2_7`

The project is fairly straight forward. 3 modules, with dependency management inside parent project. Parent project's parent is spring boot 2.4.5

**Comment by mxhc** [comment]
It works with older versions:

`mvn org.openrewrite.maven:rewrite-maven-plugin:4.36.0:run -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-spring:4.29.0  -DactiveRecipes=org.openrewrite.java.spring.boot2.UpgradeSpringBoot_2_7`

**Comment by knutwannheden** [comment]
I have also encountered this issue. In my case I think the problem was that a given dependency could not be resolved. This may be different from your use case, but I think it would be nice if also projects with unresolved dependencies could optionally be processed using a "best effort" (similarly to how a Java source with unresolved references could be processed if the corresponding dependency were completely missing in the `pom.xml` file.

**Comment by tkvangorder** [comment]
Thanks for reporting @mxhc , 

Can you provide the full stack trace by running maven with the `-e` option? 

Thank you!

**Comment by tkvangorder** [comment]
This has been fixed with [#2497](https://github.com/openrewrite/rewrite/pull/2497)
